pvanalytics/system.py

```diff
@@ -1,10 +1,283 @@
+            if quadratic_median < median_r2_min:
+                return Tracker.UNKNOWN
+        elif system_tracking is Tracker.TRACKING:
+            quartic_median = _fit.quartic_restricted_r2(
+                x=median.index,
+                y=median,
+                noon=middle
+            )
+            if quartic_median < median_r2_min:
+                return Tracker.UNKNOWN
+    return system_tracking
+
+
+def is_tracking_envelope(series, daytime, clipping, clip_max=0.1,
+                         envelope_quantile=0.995, fit_median=True,
+                         median_r2_min=0.9, fit_params=None,
+                         envelope_min_fraction=0.05,
+                         median_min_fraction=0.025,
+                         seasonal_split=None):
+    """Infer whether the system is equipped with a tracker.
+
+    Data is grouped by minute of the day and a maximum power or
+    irradiance envelope (the `envelope_quantile` value at each minute)
+    is calculated. Quadratic and quartic curves are fit to this daily
+    envelope and the :math:`r^2` of the curve fits are used determine
+    whether the system is tracking or fixed.
+
+    If the quadratic fit is a sufficiently good, then
+    :py:const:`Tracker.FIXED` is returned.
+
+    If the quartic fit is sufficiently good fit and the quadratic fit
+    is sufficiently bad, then :py:const:`Tracker.TRACKING` is
+    returned.
+
+    If neither fit is sufficiently good then
+    :py:const:`Tracker.UNKNOWN` is returned.
+
+    Optionally, an additional fit is made to the median of the
+    data at each minute to confirm the determination of tracking
+    or fixed. If performed, tracking or fixed is judged from the fit
+    to the median and this result must be consistent with the fit
+    to the upper envelope. If not, :py:const:`Tracker.UNKNOWN`
+    is returned.
+
+    Parameters
+    ----------
+    series : Series
+        Timezone localized Series of power or irradiance data.
+    daytime : Series
+        Boolean Series with True for times that are during the day.
+    clipping : Series
+        Boolean Series identifying where power or irradiance is being
+        clipped.
+    clip_max : float, default 0.1
+        If the fraction of data flagged as clipped is greater than
+        `clip_max` then it cannot be determined whether the system is
+        tracking or fixed and :py:const:`Tracker.UNKNOWN` is returned.
+    envelope_quantile : float, default 0.995
+        Quantile used to determine the upper power or irradiance
+        envelope.
+    fit_median : boolean, default True
+        Perform a secondary fit with the median power or irradiance to
+        validate that the profile is consistent through the entire
+        data set.
+    median_r2_min : float, default 0.9
+        Minimum :math:`r^2` for a curve fit to the median power or
+        irradiance at each minute of the day (Applies only if
+        `fit_median` is True).
```

Comment:
Editorial preference: group the arguments related to the median fit together, after the arguments for the envelope fit.